packages/o-spreadsheet-engine/src/functions/module_lookup.ts

```diff
@@ -1104,3 +1114,207 @@ export const OFFSET = {
+// -----------------------------------------------------------------------------
+
+export const ARRAYTOTEXT = {
+  description: _t(
+    "returns an array of text values from any specified range. It passes text values unchanged, and converts non-text values to text."
+  ),
+  args: [
+    arg("array (range)", _t("The array to convert into text")),
+    arg("format (number, default=0)", _t("The format of the returned data."), FORMAT_OPTIONS),
+  ],
+  compute: function (
+    array: Matrix<{ value: string }>,
+    format: Maybe<FunctionResultObject> = { value: 0 }
+  ) {
+    let result = "";
+    const symb = { 0: ",", 1: ";" };
+    const _format = toNumber(format, this.locale);
+    if (!(_format in Object.keys(symb))) {
+      return new EvaluationError(_t("Format must be 0 or 1"));
+    }
+    for (let row = 0; row < array[0].length; row++) {
+      for (let col = 0; col < array.length; col++) {
+        const value = array[col][row].value;
+        let text: string;
+        if (isEvaluationError(value)) {
+          text = value;
+        } else if (typeof value === "string") {
+          text = _format ? `"${value}"` : value;
+        } else {
+          text = toString(value);
+        }
+        result = result + text;
+        if (col !== array.length - 1) {
+          result = result + ",";
+        }
+      }
+      if (row !== array[0].length - 1) {
+        result = result + symb[_format];
+      }
+    }
+    if (_format === 1) {
+      result = "{" + result + "}";
+    }
+    return result;
+  },
+  isExported: true,
+} satisfies AddFunctionDescription;
+
+// -----------------------------------------------------------------------------
+// FORMULATEXT
+// -----------------------------------------------------------------------------
+
+export const FORMULATEXT = {
+  description: _t("Returns a formula as a string."),
+  args: [arg("cell_reference (meta)", _t("A reference to a cell."))],
+  compute: function (cellReference: { value: string }) {
+    const { sheetName, xc } = splitReference(cellReference.value);
+    const { col, row } = toCartesian(xc);
+    const sheetId: UID =
+      (sheetName && this.getters.getSheetIdByName(sheetName)) ?? this.getters.getActiveSheetId();
+    const result = this.getters.getCell({ sheetId, col, row });
+    if (result && isFormula(result.content)) {
+      return result.content;
+    } else {
+      return new NotAvailableError(_t("The cell does not contain a formula."));
+    }
+  },
+  isExported: true,
+} satisfies AddFunctionDescription;
+
+// -----------------------------------------------------------------------------
+// TAKE
+// -----------------------------------------------------------------------------
+
+export const TAKE = {
+  description: _t(
+    "Returns a specified number of contiguous rows or columns from the start or end of an array."
+  ),
+  args: [
+    arg("array (range)", _t("The array from which to take rows or columns.")),
+    arg(
+      "rows (number)",
```

Comment:
it is optional, right ?
```suggestion
      "rows (number, optional)",
```